kornia/nerf/nerf_solver.py

```diff
@@ -0,0 +1,201 @@
+        log_space_encoding=True,
+        lr: float = 1.0e-3,
+    ) -> None:
+        r"""Initializes training.
+
+        Args:
+            cameras: Scene cameras in the order of input images: PinholeCamera
+            min_depth: sampled rays minimal depth from cameras: float
+            max_depth: sampled rays maximal depth from cameras: float
+            ndc: convert ray parameters to normalized device coordinates: bool
+            imgs: Scene 2D images (one for each camera): Images
+            num_img_rays: Number of rays to randomly cast from each camera: math: `(B)`
+            batch_size: Number of rays to sample in a batch: int
+            num_ray_points: Number of points to sample along rays: int
+            irregular_ray_sampling: Whether to sample ray points irregularly: bool
+            log_space: Whether frequency sampling should be log spaced: bool
+            lr: Learning rate: float
+        """
+        # self._check_camera_image_consistency(cameras, imgs)   # FIXME: Check if required
+
+        self._cameras = cameras
+        self._min_depth = min_depth
+        self._max_depth = max_depth
+        self._ndc = ndc
+
+        self._imgs = imgs
+        self._num_img_rays = None
+        if isinstance(num_img_rays, int):
+            self._num_img_rays = torch.tensor([num_img_rays] * cameras.batch_size)
+        elif torch.is_tensor(num_img_rays):
+            self._num_img_rays = num_img_rays
+        elif num_img_rays is not None:
+            raise TypeError('num_img_rays can be either an int or a torch.tensor')
+
+        self._batch_size = batch_size
+
+        self._nerf_model = NerfModel(
+            num_ray_points, irregular_ray_sampling=irregular_ray_sampling, log_space_encoding=log_space_encoding
+        )
+        self._nerf_model.to(device=self._device, dtype=self._dtype)
+        self._opt_nerf = optim.Adam(self._nerf_model.parameters(), lr=lr)
+
+    @property
+    def nerf_model(self) -> nn.Module:
+        return self._nerf_model
+
+    # FIXME: Remove this function - consistency is checked in
+    @staticmethod
+    def _check_camera_image_consistency(cameras: PinholeCamera, imgs: Images):
+        if cameras is None:
+            raise ValueError('Invalid camera object')
+        if imgs is None:
+            raise ValueError('Invalid image list object')
+        if cameras.batch_size != len(imgs):
+            raise ValueError('Number of cameras must match number of input images')
+        if not all(img.shape[0] == 3 for img in imgs):
+            raise ValueError('All images must have three RGB channels')
+        if not all(height == img.shape[1] for height, img in zip(cameras.height.tolist(), imgs)):
+            raise ValueError('All image heights must match camera heights')
+        if not all(width == img.shape[2] for width, img in zip(cameras.width.tolist(), imgs)):
+            raise ValueError('All image widths must match camera widths')
+
+    def _train_one_epoch(self) -> float:
```

Comment:
can we target as one of next steps to update `NerfSolver` using the `kornia.Trainer` (and updating whatever is necessary) ?